### ISSUE TITLE
Roboelectric with AdView

### ISSUE THREAD

**Issue by refaelsh** [robolectric/robolectric#3991]
Here is my Activity:

    override fun onCreate(savedInstanceState: Bundle?)
    {
        .
        .
        .
        MobileAds.initialize(this, "ca-app-pub-8453872223973370/8999449063")
        .
        .
        .
        val adRequest = AdRequest.Builder().addTestDevice("FA27DA12E82D0830F178ACC8BDDC9FBC").build()
        adView.loadAd(adRequest)
        .
        .
        .
        Fabric.with(this, Crashlytics())
    }

Here is my test:

        @RunWith(RobolectricTestRunner::class)
        class MyMapFragmentTest
        {
            @Test
            fun clickingLogin_shouldStartLoginActivity()
            {
                val context: MainActivity = Robolectric.setupActivity(MainActivity::class.java)
            }
        }

Test fails with the following stack trace:

        java.lang.UnsupportedOperationException: sorry, not supporting public abstract boolean 
        android.content.pm.IPackageManager.isInstantApp(java.lang.String,int) throws 
        android.os.RemoteException yet!
	at org.robolectric.shadows.ShadowActivityThread$1.invoke(ShadowActivityThread.java:55)
	at com.sun.proxy.$Proxy15.isInstantApp(Unknown Source)
	at android.app.ApplicationPackageManager.isInstantApp(ApplicationPackageManager.java:832)
	at android.app.ApplicationPackageManager.isInstantApp(ApplicationPackageManager.java:826)
	at com.google.android.gms.common.wrappers.InstantApps.isInstantApp(Unknown Source)
	at com.google.android.gms.common.wrappers.PackageManagerWrapper.isCallerInstantApp(Unknown Source)
	at com.google.android.gms.ads.internal.zzd.zza(Unknown Source)
	at com.google.android.gms.ads.internal.zzd.zza(Unknown Source)
	at com.google.android.gms.ads.internal.zzd.zza(Unknown Source)
	at com.google.android.gms.ads.internal.zza.zzb(Unknown Source)
	at com.google.android.gms.ads.internal.zzy.zzb(Unknown Source)
	at com.google.android.gms.internal.ads.zzly.zza(Unknown Source)
	at com.google.android.gms.ads.BaseAdView.loadAd(Unknown Source)
	at com.google.android.gms.ads.AdView.loadAd(Unknown Source)
	at com.dev.jerusalemdirection.MainActivity.onCreate(MainActivity.kt:54)
	at android.app.Activity.performCreate(Activity.java:7009)
	at android.app.Activity.performCreate(Activity.java:7000)
	at org.robolectric.util.ReflectionHelpers.callInstanceMethod(ReflectionHelpers.java:249)
	at org.robolectric.android.controller.ActivityController$1.run(ActivityController.java:62)
	at org.robolectric.shadows.ShadowLooper.runPaused(ShadowLooper.java:359)
	at org.robolectric.android.controller.ActivityController.create(ActivityController.java:59)
	at org.robolectric.android.controller.ActivityController.create(ActivityController.java:69)
	at org.robolectric.android.controller.ActivityController.setup(ActivityController.java:176)
	at org.robolectric.Robolectric.setupActivity(Robolectric.java:88)
	at com.dev.jerusalemdirection.MyMapFragmentTest.clickingLogin_shouldStartLoginActivity(unit_tester.kt:19)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:570)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:253)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:130)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:42)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:84)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
	at com.intellij.rt.execution.application.AppMainV2.main(AppMainV2.java:131)

When I remove from `onCreate` everything AdMobs related, this exception goes away.
How can I use Roboeletric on activity with AdMobs?

Thanks. Refael.

**Comment by jaredsburrows** [comment]
Try using `buildActivity` instead of `setupActivity ` with a controller. You will have more control as well.

With Robolectric 4.0:
```
@RunWith(AndroidJUnit4.class)
public class MainActivityTest {
    @Test public void testOnCreateNotNull() {
        ActivityController<MainActivity> controller = Robolectric.buildActivity(MainActivity.class);
        Activity activity = controller.create().destroy().get();

        assertThat(activity).isNotNull();
    }
}
```

**Comment by refaelsh** [comment]
I will try it. Should work. Thank you! :-)

**Comment by jongerrish** [comment]
It fails because we are not shadowing PackageManager.isInstantApp()

Shouldn't be too hard for us to add this, will try and get a fix into 4.0.2

**Comment by jongerrish** [comment]
I have a fix for this and it should be in 4.0.2

Will just return false for now until we figure out how we want to configure it, but at least your test won't blow up.

**Comment by xian** [comment]
Moving to 4.1 since this isn't a 4.0 regression.
